R/db-sql.R

```diff
@@ -245,30 +248,20 @@ sql_query_wrap.DBIConnection <- function(con, from, name = NULL, ..., lvl = 0) {
 }
 
 sql_query_wrap_helper <- function(con, from, name, ..., lvl, as) {
-  if (is.ident(from)) {
-    out <- setNames(from, name)
-    return(out)
-  }
+  from <- as_from(from)
 
-  if (is_schema(from) || is_catalog(from) || inherits(from, "Id")) {
-    out <- setNames(as.sql(from, con), name)
+  if (is.sql(from)) {
+    from <- sql_indent_subquery(from, con, lvl)
+    # some backends, e.g. Postgres, require an alias for a subquery
+    name <- name %||% unique_subquery_name()
+    out <- glue_sql2(con, "{.sql from}", if (as) " AS", " {.name name}")
     return(out)
   }
 
-  from <- sql_indent_subquery(from, con, lvl)
-  # some backends, e.g. Postgres, require an alias for a subquery
-  name <- as_subquery_name(name)
-  glue_sql2(con, "{.sql from}", if (as) " AS", " {.name name}")
-}
-
-as_subquery_name <- function(x, default = ident(unique_subquery_name())) {
-  if (is.ident(x)) {
-    x
-  } else if (is.null(x)) {
-    default
-  } else {
-    ident(x)
+  if (identical(table_ident_name(from), name)) {
```

Comment:
What case does this catch?

Reply:
This was to prevent an unnecessary table alias. I now moved this into `escape()` where it probably makes a bit more sense.